pkg/compose/build_buildkit.go

```diff
@@ -95,16 +95,16 @@ func (s *composeService) getDrivers(ctx context.Context) ([]build.DriverInfo, er
 	dis := make([]build.DriverInfo, len(ng.Nodes))
 	var f driver.Factory
 	if ng.Driver != "" {
-		factories := driver.GetFactories()
+		factories := driver.GetFactories(true)
 		for _, fac := range factories {
 			if fac.Name() == ng.Driver {
 				f = fac
 				continue
 			}
 		}
 		if f == nil {
-			if f = driver.GetFactory(ng.Driver, true); f == nil {
-				return nil, fmt.Errorf("failed to find buildx driver %q", ng.Driver)
+			if f, err = driver.GetFactory(ng.Driver, true); f == nil || err != nil {
+				return nil, fmt.Errorf("failed to find buildx driver %q, error: %s", ng.Driver, err.Error())
```

Comment:
```suggestion
			if f, err = driver.GetFactory(ng.Driver, true); f == nil || err != nil {
				return nil, fmt.Errorf("failed to find buildx driver %q, error: %v", ng.Driver, err)
```

Minor: I assume it's an invariant that `f != nil` if `err == nil` so this is probably excessively defensive